### PR TITLE
Add user lookup API

### LINKED ISSUE #1457: Expose User Lookup API
### Goal

Our users need a way to look up the random generated user IDs to connect those with their own user models of existing backends and can interact with user-specific Realms. We've an API which covers a minimum useful solution to achieve a user lookup. As this is subject to change, it would be desirable to wrap this API endpoint, so that our users can rely on a stable API and we can change the implementation details.

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Fixes #1457 

##  TODO

* [x] Changelog entry
* [x] Tests (if applicable)
* [x] Update PCL (if applicable)

### CODE REVIEW

File: Platform.PCL/Realm.Sync.PCL/UserPCL.cs [hunk 1]
```diff
@@ -202,6 +202,27 @@ public Task ChangePasswordAsync(string userId, string newPassword)
             return null;
         }
 
+        /// <summary>
+        /// Looks up user's information by provider id. This is useful when you know the id of a user in a provider's system,
+        /// e.g. on Facebook and want to find the associated Realm user's Id.
+        /// </summary>
+        /// <param name="provider">The provider that the user has signed up with.</param>
+        /// <param name="providerId">The id of the user in the provider's system.</param>
```
Comment (kristiandupont) [inline Platform.PCL/Realm.Sync.PCL/UserPCL.cs]: I hate this name. Is it publicly exposed in the other SDK's?

Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/UserPCL.cs]: In js it's `provider, username`, in Java, it's `provider, providerId`.

Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/UserPCL.cs]: I'm struggling to find a better name, but I'm definitely open for suggestions!

**Comment by bdash** [comment]
Did y'all take a look at what the Java and JavaScript versions of these APIs are like?

**Comment by nirinchev** [comment]
I did, why?

File: Realm/Realm.Sync/Dtos/UserInfo.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace Realms.Sync
     /// <summary>
     /// The response of a user lookup operation.
```
Comment (fealebenpae) [inline Realm/Realm.Sync/Dtos/UserInfo.cs]: The comment could use updating now that the class is renamed.

**Comment by nirinchev** [comment]
CI failures will be resolved by #1489.

**Pull request merged**